// ant-design-x-vue/es/index.mjs
import { defaultPrefixCls as o, default as r } from "./x-provider/hooks/use-x-provider-context.mjs";
import { _ as f } from "./x-provider/index2.mjs";
import { default as u } from "./bubble/index.mjs";
import { default as l } from "./conversations/Conversations.mjs";
import { default as x } from "./welcome/Welcome.mjs";
import { default as n } from "./prompts/Prompts.mjs";
import { default as h } from "./suggestion/Suggestion.mjs";
import { default as g } from "./sender/index.mjs";
import { default as b } from "./attachments/index.mjs";
import { default as c } from "./thought-chain/ThoughtChain.mjs";
import { default as A } from "./x-stream/x-stream.mjs";
import { default as B } from "./x-request/x-fetch.mjs";
import { default as F } from "./x-request/x-request.mjs";
import { XAgent as R, default as T } from "./use-x-agent/use-x-agent.mjs";
import { default as _ } from "./use-x-chat/use-x-chat.mjs";
import { theme as k } from "ant-design-vue";
import { default as y } from "./bubble/BubbleList.mjs";
export {
  b as Attachments,
  u as Bubble,
  y as BubbleList,
  l as Conversations,
  n as Prompts,
  g as Sender,
  h as Suggestion,
  c as ThoughtChain,
  x as Welcome,
  R as XAgent,
  B as XFetch,
  f as XProvider,
  F as XRequest,
  A as XStream,
  o as defaultPrefixCls,
  k as theme,
  T as useXAgent,
  _ as useXChat,
  r as useXProviderContext
};
